### ISSUE TITLE
A check on my Results

### ISSUE THREAD

**Issue by poaboagye** [GraphPKU/PiSSA#6]
Hi, thanks for this great work. Please. I have a quick question for you. I just finished running train.py on Llama-2-7b-hf and MATH_inference.py with these arguments:

python train.py \
    --model_name_or_path meta-llama/Llama-2-7b-hf \
    --data_path meta-math/MetaMathQA \
    --dataset_split train[:100000] \
    --dataset_field query response \
    --output_dir ./results/llama-2/meta-math \
    --init_lora_weights pissa \
    --report_to wandb \
    --merge_and_save True \
    --bf16 True \
    --num_train_epochs 1 \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 64 \
    --save_strategy "steps" \
    --save_steps 1000 \
    --save_total_limit 1 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True

python inference/gsm8k_inference.py --model ./results/llama-2/meta-math
python inference/MATH_inference.py --model ./results/llama-2/meta-math

Then I evaluated it on the MATH dataset (MATH_test.jsonl) and this was my result. With an accuracy of about 5.8%. I think Pissa was 7.44% in the paper. Is there something I am not doing right to get the same result? :

start=== 0 , end==== 9223372036854775807
length==== 5000 , acc==== 0.0584

Here is the training report:

https://api.wandb.ai/links/poaboagye/qt7iw390

**Comment by fxmeng** [comment]
Thank you for your interest to PiSSA. 
I noticed that you did not set the rank of PiSSA in the script; hence, the default rank=16 was used. However, the accuracy of 7.44% on the Math dataset reported in the paper was obtained under the condition of rank=128. The paper indicates that the fine-tuning performance of Llama-2 improves as the number of trainable parameters increases. You can modify this hyperparameter and redo the experiments.

**Comment by poaboagye** [comment]
Thank you for your swift response. I will have a look at that.

Prince
